ui/src/callWithTelemetryAndErrorHandling.ts

```diff
@@ -69,6 +70,7 @@ function handleError(context: IActionContext, callbackId: string, error: any): v
         context.properties.result = 'Failed';
         context.properties.error = errorData.errorType;
         context.properties.errorMessage = errorData.message;
+        context.properties.stack = errorData.stack;
```

Comment:
I've got mixed feelings about adding the call stack to telemetry for errors. It can be useful, but will increase the data we send (for errors only), e.g.:

** TELEMETRY("undefined/callbackId", undefined) properties={"isActivationEvent":"false","cancelStep":"","result":"Failed","stack":"Error: testFuncErrorAsync\n    at /Users/stephwe/repos/vscode-azuretools/ui/test/callWithTelemetryAndErrorHandling.test.ts:24:11\n    at Generator.next (<anonymous>)\n    at __awaiter (/Users/stephwe/repos/vscode-azuretools/ui/out/test/callWithTelemetryAndErrorHandling.test.js:11:71)\n    at new Promise (<anonymous>)\n    at __awaiter (/Users/stephwe/repos/vscode-azuretools/ui/out/test/callWithTelemetryAndErrorHandling.test.js:7:12)\n    at testFuncErrorAsync (/Users/stephwe/repos/vscode-azuretools/ui/out/test/callWithTelemetryAndErrorHandling.test.js:30:12)\n    at Object.<anonymous> (/Users/stephwe/repos/vscode-azuretools/ui/test/callWithTelemetryAndErrorHandling.test.ts:62:24)\n    at Generator.next (<anonymous>)\n    at __awaiter (/Users/stephwe/repos/vscode-azuretools/ui/out/test/callWithTelemetryAndErrorHandling.test.js:11:71)\n    at new Promise (<anonymous>)\n    at __awaiter (/Users/stephwe/repos/vscode-azuretools/ui/out/test/callWithTelemetryAndErrorHandling.test.js:7:12)\n    at Object.<anonymous> (/Users/stephwe/repos/vscode-azuretools/ui/out/test/callWithTelemetryAndErrorHandling.test.js:64:24)\n    at Object.<anonymous> (/Users/stephwe/repos/vscode-azuretools/ui/src/callWithTelemetryAndErrorHandling.ts:53:59)\n    at Generator.next (<anonymous>)\n    at __awaiter (/Users/stephwe/repos/vscode-azuretools/ui/out/src/callWithTelemetryAndErrorHandling.js:11:71)\n    at new Promise (<anonymous>)\n    at __awaiter (/Users/stephwe/repos/vscode-azuretools/ui/out/src/callWithTelemetryAndErrorHandling.js:7:12)\n    at Object.callWithTelemetryAndErrorHandling (/Users/stephwe/repos/vscode-azuretools/ui/out/src/callWithTelemetryAndErrorHandling.js:56:12)\n    at Object.<anonymous> (/Users/stephwe/repos/vscode-azuretools/ui/test/callWithTelemetryAndErrorHandling.test.ts:60:31)\n    at Generator.next (<anonymous>)\n    at __awaiter (/Users/stephwe/repos/vscode-azuretools/ui/out/test/callWithTelemetryAndErrorHandling.test.js:11:71)\n    at new Promise (<anonymous>)\n    at __awaiter (/Users/stephwe/repos/vscode-azuretools/ui/out/test/callWithTelemetryAndErrorHandling.test.js:7:12)\n    at assertThrowsAsync_1.assertThrowsAsync (/Users/stephwe/repos/vscode-azuretools/ui/test/callWithTelemetryAndErrorHandling.test.ts:60:24)\n    at Object.<anonymous> (/Users/stephwe/repos/vscode-azuretools/ui/test/assertThrowsAsync.ts:15:15)\n    at Generator.next (<anonymous>)\n    at __awaiter (/Users/stephwe/repos/vscode-azuretools/ui/out/test/assertThrowsAsync.js:11:71)\n    at new Promise (<anonymous>)\n    at __awaiter (/Users/stephwe/repos/vscode-azuretools/ui/out/test/assertThrowsAsync.js:7:12)\n    at Object.assertThrowsAsync (/Users/stephwe/repos/vscode-azuretools/ui/out/test/assertThrowsAsync.js:20:12)\n    at Object.<anonymous> (/Users/stephwe/repos/vscode-azuretools/ui/test/callWithTelemetryAndErrorHandling.test.ts:59:15)\n    at Generator.next (<anonymous>)\n    at fulfilled (/Users/stephwe/repos/vscode-azuretools/ui/out/test/callWithTelemetryAndErrorHandling.test.js:8:58)","error":"Error","errorMessage":"testFuncErrorAsync"}, measures={"duration":0.001}

Reply:
One option would be to limit to 100 characters or something

Reply:
I like setting a limit on the call stack. Could be number of characters or could be number of lines